monkey/common/utils/file_utils.py

```diff
@@ -1,7 +1,27 @@
 import hashlib
+import logging
 import os
+import platform
+import stat
+from contextlib import contextmanager
 from pathlib import Path
-from typing import BinaryIO, Iterable
+from typing import BinaryIO, Generator, Iterable
+
+
+def is_windows_os() -> bool:
+    return platform.system() == "Windows"
+
+
+if is_windows_os():
+    import win32file
+    import win32job
+    import win32security
+
+    from . import windows_permissions
```

Comment:
Secure file/directories could probably be a separate package/file. Creating secure directories/files has a lot of logic and hacks like these. It's always better when packages are smaller and do less things, because now you read these strange imports and also have to skip some code until they become relevant.